Eastern-Research-Group/mywaterway
# Feature/reporting cycle changes

## Related Issues:
* https://app.breeze.pm/projects/100762/cards/3414821
* https://app.breeze.pm/projects/100762/cards/3420902

## Main Changes:
* Fixed an issue with popups being empty on the waterbody report page if you click the "View On Map" button.
* Updated the waterbody report page to take an optional reportingCycle parameter. If the reporting cycle parameter is provided then the assessments service will be called with the reporting parameter provided in the URL. If the reporting cycle parameter is not provided then the app will determine the reporting cycle from the GIS data (if available) and use that to call the assessments service.
* On the waterbody report page, added an info box with a link to the most current waterbody report data. This new info box and link are only displayed if the reporting cycle in the GIS data is later than the reporting cycle provided in the url. 
* Updated the community page to pass the reporting cycle (from GIS data) into the waterbody report page. 
* Updated the actions page to pass the reporting cycle (from GIS data) into the waterbody report page.
* Updated the state page to obtain the reporting cycle from GIS data for use in the usesStateSummary service call and for display on the site specific charts and footer of the advanced search page map. 
    * Some states like Missouri allow waterbodies to have different reporting cycles. For these scenarios the latest year is chosen
    * American Samoa and Northern Mariana Islands do not have GIS data. For this case the usesStateSummary will be called without the reporting cycle parameter.
    * Note: The reporting cycle displayed below the surveys chart is still independent.
* Fixed one of the community page tests related to the identified issues tab.
* Fixed tests related to the waterbody report links.
* Added a new test for routing to the waterbody report page with a cycle year.

## Steps To Test:
1. Navigate to http://localhost:3000/community/071401010403/overview
2. Make sure the view waterbody report links include a cycle year at the end of the url.
    * Note: The waterbodies have different reporting cycles for this location, so verify the link has the same year as the "Year Last Reported" value. 
3. Navigate to http://localhost:3000/waterbody-report/MDNR/MO_1707.02/2016.
4. Verify the new info box is displayed at the top of the page and links to the latest waterbody report for that assessment http://localhost:3000/waterbody-report/MDNR/MO_1707.02/2018
5. Navigate to http://localhost:3000/plan-summary/21GAEPD/10018
6. Click a "View On Map" button and verify the popup has content.
7. Click on a waterbody on the map and verify the popup has content.
8. Verify that all of the waterbodies listed have the "View Waterbody Report" link. 
9. For waterbodies that have a "View On Map" button, verify the "View Waterbody Report" link has the cycle year parameter. 
10. For waterbodies that do not have a "View On Map" button, verify the "View Waterbody Report" link does not have the cycle year parameter. 
11. Open the Chrome dev tools
12. Navigate to http://localhost:3000/state/FL/water-quality-overview
13. Verify the usesStateSummary service was called with a cycle year.
14. Click "Aquatic Life" and select "Rivers and Streams"
15. Verify the reporting cycle under the pie chart is 2020 and the reporting cycle under the site specific bar chart is 2012. 
16. Navigate to http://localhost:3000/state/AS/water-quality-overview
17. Verify the usesStateSummary service was called without a cycle year.
